Controller/Admin/TransactionTabPostProcessing.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace Wirecard\Oxid\Controller\Admin;
+
+use Wirecard\Oxid\Model\Transaction;
+use Wirecard\Oxid\Controller\Admin\TransactionTab;
+
+use OxidEsales\Eshop\Core\Registry;
+use Exception;
+
+/**
+ * Controls the view for the post-processing tab.
+ */
+class TransactionTabPostProcessing extends TransactionTab
+{
+    protected $_sThisTemplate = 'transaction_tab_pp.tpl';
+
+    protected $_i18n = null;
+
+    protected $_config = null;
+
+    /**
+     * TransactionTabPostProcessing constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->_i18n = Registry::getLang();
+        $this->_config = Registry::getConfig();
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return string
+     */
+    public function render(): string
+    {
+        $template = parent::render();
+        $transaction = $this->_aViewData['transaction'];
+        $actions = $this->_getPostProcessingActions($transaction);
+        $alert = [];
+
+        try {
+            $this->validateRequest($transaction);
+
+            // if an action was requested, run its callback
+            foreach ($actions as $key => $action) {
+                if ($this->_config->getRequestParameter($key)) {
+                    $alert['message'] = call_user_func([$this, $action['callback']], $transaction);
+                    $alert['type'] = 'success';
+
+                    break;
+                }
+            }
+        } catch (Exception $e) {
+            $alert['message'] = $e->getMessage();
+            $alert['type'] = 'error';
+        }
+
+        $this->_aViewData['actions'] = $actions;
+        $this->_aViewData['alert'] = $alert;
+        $this->_aViewData['currency'] = $transaction->wdoxidee_ordertransactions__wdoxidee_currency->value;
+        $this->_aViewData['requestParameters'] = $this->getRequestParameters();
+
+        return $template;
+    }
+
+    /**
+     * Returns an array of current request parameters.
+     *
+     * @return array
+     */
+    public function getRequestParameters(): array
+    {
+        return [
+            'amount' => $this->_config->getRequestParameter('amount'),
+        ];
+    }
+
+    /**
+     * Validates the current request.
+     *
+     * @param Transaction $transaction
+     * @return void
+     */
+    public function validateRequest(Transaction $transaction): void
+    {
+        $requestParameters = $this->getRequestParameters();
+        $amount = $requestParameters['amount'];
+
+        if ($amount === null) {
+            return;
```

Comment:
Why is there no need to through an **exception** in case the amount equals null?

Reply:
The amount is `null` when the view is displayed without submitting the form.